### ISSUE TITLE
FFME randomly crashes with DirectSound Audio Driver

### ISSUE THREAD

**Issue by Dartjek** [unosquare/ffmediaelement#388]
# FFME randomly crashes while you are move to another position on timeline, quickly shake and etc.

Unfortunately, It's rare reproduced. Sometimes FFME just randomly crashes while you are making randomly move to different parts of video.

With enabled collecting user-mode dumps the player generates theirs. The problem occurs around dsound.dll (You will see it at the call stack window). 

Usual reasons of crash: 
* heap was corrupted
* Stack cookie instrumentation code detected a stack-based buffer overrun
* Access violation reading location

## Issue Categories

 - [X] Bug

## Version Information

 - [X] NuGet Package *3.1.280 - 3.1.300*
 - [X] Build From *Master* branch, Commit 29b36704

## Steps to Reproduce
Here are the simplest way that I've found to reproduce bug with similar reasons the crash.

1. Enable [Collecting User-Mode Dumps](https://docs.microsoft.com/en-us/windows/desktop/wer/collecting-user-mode-dump)
2. Compile Windows.Sample and load video
3. Quickly shake the thumb along timeline during a few seconds
4. Just close the FFME windows sample.
5. You will see a generated dump at the folder that you point in the dump settings.

## Additional Info
[Media info](https://github.com/unosquare/ffmediaelement/files/3173075/Test_zapis_03.txt) of the file that was used to reproduce the bug
[Crash dump](https://1drv.ms/u/s!Avfq51Q7jyMMrEtWb4zauyO5L1E3) on one drive

**Comment by mariodivece** [comment]
Have you tried using Legacy Audio? Could you try it out and let me know? If crashes don't occur when legacy audio out is enabled, than it means I am doing something wrong with DirectSound...

https://github.com/unosquare/ffmediaelement/blob/c17fca9836880725f071ce26f863548597708487/Unosquare.FFME.Windows/Common/RendererOptions.cs#L30

**Comment by Dartjek** [comment]
I've tried it for several days and I have no crashes or crashdumps. Seems it works.

**Comment by adamhewitt627** [comment]
After days of debugging with little stack traces, I finally found my crash related to "runtime callable wrappers" - which led me to this item. (I had also updated from Framework to .NET6, so it wasn't obviously an FFME failure until now)

Switching to the legacy audio output works, but it would be nice if this could be reopened and eventually fixed. I may try to fix it myself on my fork, and would pick it back over here if I can.

The crash is specifically during the garbage collector, so it looks like a misuse of the COM objects and releasing resources deterministically.
